src/llir/raise/early.rs

```diff
@@ -238,26 +245,49 @@ fn decode_args_with_abi(
                 ScalarValue::Int(extra_arg as _)
             },
 
-            | ArgEncoding::Integer { arg0: false, size: 4, ty_color: _ }
-            | ArgEncoding::Color
             | ArgEncoding::JumpOffset
             | ArgEncoding::JumpTime
-            | ArgEncoding::Padding
+            | ArgEncoding::Integer { arg0: false, size: 4, format: ast::IntFormat { unsigned: false, radix: _ }, .. }
             => {
                 decrease_len(emitter, &mut remaining_len, 4)?;
-                ScalarValue::Int(args_blob.read_u32().expect("already checked len") as i32)
+                ScalarValue::Int(args_blob.read_i32().expect("already checked len") as i32)
```

Comment:
Oops, how silly.

The `as` cast can be removed then.

```suggestion
                ScalarValue::Int(args_blob.read_i32().expect("already checked len"))
```